examples/run/alpaka/throughput_mt.cpp

```diff
@@ -5,20 +5,17 @@
  * Mozilla Public License Version 2.0
  */
 
-// Project include(s).
-#include "traccc/alpaka/utils/get_vecmem_resource.hpp"
-
 // Local include(s).
 #include "../common/throughput_mt.hpp"
+
 #include "full_chain_algorithm.hpp"
 
 int main(int argc, char* argv[]) {
 
     // Execute the throughput test.
     static const bool use_host_caching = true;
-    return traccc::throughput_mt<
-        traccc::alpaka::full_chain_algorithm,
-        traccc::alpaka::vecmem_resources::host_memory_resource>(
+    return traccc::throughput_mt<traccc::alpaka::full_chain_algorithm,
+                                 vecmem::host_memory_resource>(
```

Comment:
This is the only bit I'm still not 100% sure on the impact of.

I.e. we now always use the default `vecmem::host_memory_resource`, rather than `vecmem::cuda::host_memory_resource` etc as appropriate.

I guess there is some performance impact around missing out on the pinning the going via `cudaMallocHost` gives? Not sure on the most appropriate way to surface this type though, or if it is easier to change the interface instead.